generated/nidcpower/nidcpower/ni_lcr_measurement.py

```diff
@@ -0,0 +1,242 @@
+                if field_name == "measurement_mode":
+                    field_value = enums.InstrumentMode(field_value).value
+                setattr(self, field_name, field_value)
+        else:
+            for field_name, _, _ in _ni_lcr_measurement_field_metadata:
+                field_value = locals()[field_name]
+                if field_name == "measurement_mode":
+                    field_value = enums.InstrumentMode(field_value).value
+                setattr(self, field_name, field_value)
+
+    def __repr__(self):
+        field_value_strings = []
+        for field_name, _, _ in _ni_lcr_measurement_field_metadata:
+            field_value = getattr(self, field_name)
+            field_value_strings.append("{0}={1}".format(field_name, field_value))
+        return "{0}(data=None, {1})".format(self.__class__.__name__, ", ".join(field_value_strings))
+
+    def __str__(self):
+        max_field_label_len = max(
+            len(field_label)
+            for _, field_label, _ in _ni_lcr_measurement_field_metadata
+            if field_label is not None
+        )
+        field_value_strings = []
+        for field_name, field_label, field_ctype in _ni_lcr_measurement_field_metadata:
+            if field_label is not None and not field_name.endswith("_imaginary"):
+                # Determines row_format
+                if field_ctype == nidcpower._visatype.ViReal64:
+                    row_format = "{{:<{}}}: {{:,.6g}}\n".format(max_field_label_len)
+                else:
+                    row_format = "{{:<{}}}: {{:}}\n".format(max_field_label_len)
+                # Process field_value if necessary
+                if field_name.endswith("_real"):
+                    field_value = complex(
+                        getattr(self, field_name),
+                        getattr(self, field_name.replace("_real", "_imaginary")),
+                    )
+                else:
+                    field_value = getattr(self, field_name)
+                    if field_name == "measurement_mode":
+                        field_value = enums.InstrumentMode(field_value).name
+                    elif field_name in ("dc_in_compliance", "ac_in_compliance", "unbalanced"):
+                        # Compare field_name instead of field_ctype as ViBoolean is an alias of
+                        #  ViUInt16
+                        field_value = bool(field_value)
+                # Generate string
+                field_value_strings.append(row_format.format(field_label, field_value))
+        return "".join(field_value_strings)
+
+
+class NILCRMeasurement(object):
+    def __init__(
+        self,
+        data=None,
+        vdc=0.0,
+        idc=0.0,
+        stimulus_frequency=0.0,
+        ac_voltage=complex(),
+        ac_current=complex(),
+        z=complex(),
+        z_magnitude=0.0,
+        z_phase=0.0,
+        y=complex(),
+        y_magnitude=0.0,
+        y_phase=0.0,
+        ls=0.0,
```

Comment:
ls, cs, etc... are those obvious to users?

There is definitely a lack of documentation for this class that you need to address.

Reply:
`ls`, `cs`, `rs`, etc. are named based on the `nidcpower.h` header file:
```
// Z values
NIComplexNumber Z;
ViReal64 ZMagnitude;
ViReal64 ZPhase; // degrees
// Y values
NIComplexNumber Y;
ViReal64 YMagnitude;
ViReal64 YPhase; // degrees
```

I do agree that this class requires documentation. I have added documentation for all its non-reserved properties (similar documentation will also be added to the new LCR-related functions that use this custom type in #1704)